src/msg/Messenger.cc

```diff
@@ -21,7 +21,7 @@ Messenger *Messenger::create_client_messenger(CephContext *cct, std::string lnam
 uint64_t Messenger::get_pid_nonce()
 {
   uint64_t nonce = getpid();
-  if (nonce == 1) {
+  if (nonce <= 2 || getenv("CEPH_CONTAINER_HAS_INIT")) {
```

Comment:
I don't think there is a sane way to have a pid < 10 for any reasonable non-containerized environments. maybe 

```cpp
assert getenv("CEPH_CONTAINER_HAS_INIT") || nonce > 10
```

?

Reply:
I don't think we should assert, since cephadm might not be running it (could be rook, or pacific/master being deployed by octopus cephadm for upgrade (or bootstrap!).

I'll change it to 10, though.. it looks like in our case ceph-osd is ending up as pid 6 right now.